src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -182,32 +182,45 @@ def _alloc_node(self, node, requested_cores, requested_gpus,
         requested (but the call will never return more than requested).
         '''
 
-
+        # Create empty list of cores and gpus. This empty list will be populated
+        # with core ids and gpu ids if they are available in the current node.
         cores = list()
         gpus  = list()
 
+        # Count the number of free cores and gpus in the current node. This is 
+        # way quicker than actually finding the core IDs.
         free_cores = node['cores'].count(rpc.FREE)
         free_gpus  = node['gpus' ].count(rpc.FREE)
 
-        # first count the free cores/gpus, as that is way quicker than
-        # actually finding the core IDs.
+
         if partial:
-            # For partial requests the check simpliefies: we just check if we 
-            # have either, some cores *or* gpus, to serve the request
+            # For partial requests, the check simplifies: we need to check if we 
+            # have either some cores *or* gpus to serve the request, if yes we
+            # continue. 
+            # In the following line, we check for its equivalent negative logic:
+            # If both no. of cpus and no. of gpus free on this node do not satisfy the 
+            # request (i.e. there are no free cores and no free gpus), return.
+            # NOTE: This condition needs to checked only for MPI units.
```

Comment:
non-mpi CUs will never ask for partial allocations anyway - maybe we should make this clearer in the method docstring?

Reply:
Ok, happy to take the NOTE off.